packages/cf-component-icon/src/Icon.js

```diff
@@ -1,24 +1,102 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import * as svgs from './svgs.js';
+
+import { createComponent } from 'cf-style-container';
+
+const DEFAULT_HEIGHT = 15.5;
+
+const getHeight = size => {
+  let height;
+  switch (size) {
+    case '1.5x':
+      height = 23;
+      break;
+    case '2x':
+      height = 30.5;
+      break;
+    case '2.5x':
+      height = 38;
+      break;
+    case '3x':
+      height = 45.5;
+      break;
+    case '3.5x':
+      height = 53;
+      break;
+    case '4x':
+      height = 60.5;
+      break;
+    default:
+      height = DEFAULT_HEIGHT;
+  }
+  return height;
+};
+
+const getFill = (color, theme) => {
+  let fill;
+  switch (color) {
+    case 'default':
+      fill = theme.color.smoke;
+      break;
+    case 'primary':
+      fill = theme.color.marine;
+      break;
+    case 'success':
+      fill = theme.color.grass;
+      break;
+    case 'warning':
+      fill = theme.color.carrot;
+      break;
+    case 'danger':
+      fill = theme.color.apple;
+      break;
+    case 'black':
+      fill = theme.colorBlack;
+      break;
+    case 'white':
+      fill = theme.colorWhite;
+      break;
+    default:
+      fill = theme.colorBlack;
+  }
+  return fill;
+};
+
+const IconWrapper = createComponent(
+  ({ height }) => ({
+    display: 'inline-block',
+    height: height ? height : DEFAULT_HEIGHT
+  }),
+  'span',
+  ['aria-label']
+);
+
 class Icon extends React.Component {
   render() {
-    let className = `cf-icon cf-icon--${this.props.type}`;
+    const { className, size, label, color } = this.props;
+    const icon = svgs[this.props.type.replace(/-/g, '')];
```

Comment:
Not sure how I feel about this. Can't we just have our icon types match our icon names?

Reply:
I want to do that. Instead of doing a bunch of things to check for a match.